ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml

```diff
@@ -12,94 +12,79 @@ import StatusQ.Popups 0.1
 import StatusQ.Controls 0.1
 
 StatusModal {
-    id: popup
+    id: root
+
+    signal passwordChanged()
+
     width: 480
     height: 510
     closePolicy: Popup.NoAutoClose
     header.title: qsTr("Change password")
 
-    onOpened: {
-        reset();
-    }
+    onOpened: d.reset()
+
+    QtObject {
+        id: d
+
+        function lengthValidator(text) {
+            return text.length >= 6 ? "" : qsTr("At least 6 characters")
+        }
 
-    property var successPopup
-    property string indicationText: ""
-    property bool passwordInputValid
-    property bool currPasswordInputValid
-    property string currPasswordValidationError: ""
-
-    function reset() {
-        passwordInput.text = "";
-        currentPasswordInput.text = "";
-        currentPasswordInput.forceActiveFocus(Qt.MouseFocusReason);
-        popup.indicationText = "At least 6 characters. Your password protects your keys. You need it to unlock Status and transact.";
-        popup.currPasswordValidationError = "";
-        passwordInput.validationError = "";
-        popup.passwordInputValid = false;
-        popup.currPasswordInputValid = false;
+        function reset() {
+            currentPasswordInput.state = "init"
+            passwordInput.state = "init"
+            confirmPasswordInput.state = "init"
+            currentPasswordInput.forceActiveFocus(Qt.MouseFocusReason)
+        }
     }
 
     contentItem: ColumnLayout {
         id: contentItem
         anchors.fill: parent
         anchors {
-            topMargin: (Style.current.xlPadding + popup.topPadding)
+            topMargin: Style.current.xlPadding + root.topPadding
```

Comment:
just curious, why the parenthesis were removed?

Reply:
At least they aren't necessary as simple arithmetic goes left -> right

Reply:
Exactly, they weren't necessary either from an arithmetic perspective or code readability, or am I missed something? :P

Reply:
not at all! it was just curiosity :)